src/Compilers/Core/Portable/Operations/IndexedPropertyReferenceExpression_IHasArgumentsExpression.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    internal partial class IndexedPropertyReferenceExpression : IHasArgumentsExpression
+    {
+        /// <summary>
+        /// Arguments of the invocation, excluding the instance argument. Arguments are in parameter order,
+        /// and params/ParamArray arguments have been collected into arrays. Default values are supplied for
+        /// optional arguments missing in source.
+        /// </summary>
+        public ImmutableArray<IArgument> ArgumentsInParameterOrder { get; }
+        /// <summary>
+        /// Find the argument supplied for a given parameter of the target method.
+        /// </summary>
+        /// <param name="parameter">Parameter of the target method.</param>
+        /// <returns>Argument corresponding to the parameter.</returns>
+        public IArgument GetArgumentMatchingParameter(IParameterSymbol parameter)
+        {
+            throw new System.NotImplementedException();
```

Comment:
these files are just place holder file now. next PR will make these to do right thing.